### PR TITLE
ADD CBMC proof of Posix_selfTestTimerCallback function

### PR DESCRIPTION
**Pull request by Bujain**
ADD CBMC proof of Posix_selfTestTimerCallback function

Description
-----------
This PR adds files required for the CBMC proof of Posix_selfTestTimerCallback function. 

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is formatted using Uncrustify.
- [x] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/Posix_selfTestTimerCallback/selfTestTimerCallback_harness.c [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    OtaTimerCallback_t callback;
+    union sigval arg;
+
+    /* The valid range of values for OtaTimerId_t are [0,2).*/
+    __CPROVER_assume( otaTimerId >= 0 && otaTimerId < 2 );
+
+    pTimerName = ( const char * ) malloc( size * sizeof( char ) );
+
+    /* Initializing a timer callback function. */
+    callback = OtaCallback;
```
Comment (yanjos-dev) [inline test/cbmc/proofs/Posix_selfTestTimerCallback/selfTestTimerCallback_harness.c:55]: Is this always set in the code to a valid value?

File: test/cbmc/proofs/Posix_selfTestTimerCallback/selfTestTimerCallback_harness.c [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+{
+    OtaTimerId_t otaTimerId;
+    const char * pTimerName;
+    size_t size;
+    uint32_t timeout;
+    OtaTimerCallback_t callback;
+    union sigval arg;
+
+    /* The valid range of values for OtaTimerId_t are [0,2).*/
+    __CPROVER_assume( otaTimerId >= 0 && otaTimerId < 2 );
```
Comment (yanjos-dev) [inline test/cbmc/proofs/Posix_selfTestTimerCallback/selfTestTimerCallback_harness.c:50]: Is this enforced by the code?

**Review by yanjos-dev — CHANGES_REQUESTED** [review]

File: test/cbmc/proofs/Posix_selfTestTimerCallback/selfTestTimerCallback_harness.c (hunk 1 above)
Comment (Bujain) [reply test/cbmc/proofs/Posix_selfTestTimerCallback/selfTestTimerCallback_harness.c:55]: Yes. It is always the case. Whenever a timer is started the `OtaTimerCallback` in the `ota.c` source file is used as the callback function.

File: test/cbmc/proofs/Posix_selfTestTimerCallback/selfTestTimerCallback_harness.c (hunk 2 above)
Comment (Bujain) [reply test/cbmc/proofs/Posix_selfTestTimerCallback/selfTestTimerCallback_harness.c:50]: Yes. The function is passed an argument of `OtaTimerId_t` type. `OtaTimerId_t` is an enum with only 2 valid values.

**Review by yanjos-dev — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**